lib/engine/game/g_18_usa/step/buy_pullman.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+require_relative '../../../step/train'
+require_relative 'scrap_train_module'
+module Engine
+  module Game
+    module G18USA
+      module Step
+        class BuyPullman < Engine::Step::Base
+          include Engine::Step::Train
+          include ScrapTrainModule # Why would you scrap a pullman so you can buy a pullman!?
+          def actions(entity)
+            return [] if entity != current_entity
+
+            if @game.depot.upcoming.any? { |t| t.name == 'P' } && can_buy_train?(entity) && @game.pullmans_available? && \
```

Comment:
There's no reason for me to use @game.pullman_train? one line below, but not here